File: language_bnf/examples/good/reccurency.c

int sum(int x) {
    if (x < 1) {
        return 1;
    }

    int i;
    int j;
    i = 0;
    j = 0;
    while (i < x) {
        j = sum(i);
    }
    return j;
}

int main() {
    int result;

    result = sum(20);
    print(result);

    return 0;
}

File: interpreter/examples/good/reccurency.rs

fn sum(a:i32 = 1, n:i32 = 5) -> i32 {
    if n < 0 {
        return a;
    }
    return a + sum(a, n-1);
}

// function does not return a value
fn main() -> {
    print(sum(10));
}